nibabel/tests/test_arrayproxy.py

```diff
@@ -349,52 +349,139 @@ def __init__(self, *args, **kwargs):
     voxels = np.random.randint(0, 10, (10, 3))
+
+    for test in tests:
+        filetype, kfo, have_igzip, exp_persist, exp_kfo = test
+        with InTemporaryDirectory(), \
+             mock.patch('nibabel.openers.ImageOpener', CountingImageOpener), \
+             patch_indexed_gzip(have_igzip):
+            fname = 'testdata.{}'.format(filetype)
+            # create the test data file
+            if filetype == 'gz':
+                with gzip.open(fname, 'wb') as fobj:
+                    fobj.write(data.tostring(order='F'))
+            else:
+                with open(fname, 'wb') as fobj:
+                    fobj.write(data.tostring(order='F'))
+            # pass in a file name or open file handle. If the latter, we open
+            # two file handles, because we're going to create two proxies
+            # below.
+            if filetype == 'open':
+                fobj1 = open(fname, 'rb')
+                fobj2 = open(fname, 'rb')
+            else:
+                fobj1 = fname
+                fobj2 = fname
+            try:
+                proxy = ArrayProxy(fobj1, ((10, 10, 10), dtype),
+                                   keep_file_open=kfo)
+                # We also test that we get the same behaviour when the
+                # KEEP_FILE_OPEN_DEFAULT flag is changed
+                with patch_keep_file_open_default(kfo):
+                    proxy_def = ArrayProxy(fobj2, ((10, 10, 10), dtype))
+                # check internal flags
+                assert proxy._persist_opener == exp_persist
+                assert proxy._keep_file_open == exp_kfo
+                assert proxy_def._persist_opener == exp_persist
+                assert proxy_def._keep_file_open == exp_kfo
+                # check persist_opener behaviour - whether one imageopener is
+                # created for the lifetime of the ArrayProxy, or one is
+                # created on each access
+                if exp_persist:
+                    assert _count_ImageOpeners(proxy, data, voxels) == 1
+                    assert _count_ImageOpeners(proxy_def, data, voxels) == 1
+                else:
+                    assert _count_ImageOpeners(proxy, data, voxels) == 10
+                    assert _count_ImageOpeners(proxy_def, data, voxels) == 10
+                # if indexed_gzip is active, check that the file object was
+                # created correctly - the _opener.fobj will be a
+                # MockIndexedGzipFile, defined in test_openers.py
+                if filetype == 'gz' and have_igzip:
+                    assert proxy._opener.fobj._drop_handles == (not exp_kfo)
+                # if we were using an open file handle, check that the proxy
+                # didn't close it
+                if filetype == 'open':
+                    assert not fobj1.closed
+                    assert not fobj2.closed
+            except Exception:
+                print('Failed test', test)
```

Comment:
Let's remove this debug print statement. If it's useful to add context, you can use `six.raise_from` to raise a new exception with the old one as context.